app/audit.go

```diff
@@ -156,21 +125,50 @@ func (s *Server) configureAudit(adt *audit.Audit) {
 			maxQueueSize = audit.DefMaxQueueSize
 		}
 
-		filter := adt.MakeFilter(RestLevel, RestContentLevel, RestPermsLevel, CLILevel)
+		filter := adt.MakeFilter(LevelAPI, LevelContent, LevelPerms, LevelCLI)
 		formatter := adt.MakeJSONFormatter()
 		formatter.DisableTimestamp = false
 		target, err := audit.NewFileTarget(filter, formatter, opts, maxQueueSize)
 		if err != nil {
-			mlog.Error("cannot configure File audit target", mlog.Err(err))
+			errs = multierror.Append(err)
 		} else {
 			mlog.Debug("File audit target created successfully", mlog.String("filename", opts.Filename))
 			adt.AddTarget(target)
 		}
 	}
+
+	// Advanced logging for audit requires license.
+	if !bAllowAdvancedLogging || *s.Config().ExperimentalAuditSettings.AdvancedLoggingConfig == "" {
+		return errs
+	}
+	dsn := *s.Config().ExperimentalAuditSettings.AdvancedLoggingConfig
```

Comment:
Nit: If we move this up we can read the value just once.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/15076/commits/3339d015f5a56ff38398bac6188d4d396243ebb9